### ISSUE TITLE
`vue/sort-keys` ignoreGrandchildrenOf not working for descendants greater than 2

### ISSUE THREAD

**Issue by miralize** [vuejs/eslint-plugin-vue#1349]
Hopefully I'm understanding the options correctly in that if I have a component with a computed property that contains a returns object it should ignore those properties on the object returned?

My use case is that I want to only sort the data/computeds/methods etc and not touch the logic within them

Would it make sense for the rule to either accept the level at which to check or check that the depth is greater or equal to 2? 

[Permalink to line causing the issue](https://github.com/vuejs/eslint-plugin-vue/blob/master/lib/rules/sort-keys.js#L222)

**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 7
- **eslint-plugin-vue version:** 
- **Node version:** 14
- **Operating System:** macOS

**Please show your full configuration:** (sort-keys is defined in the overrides only)

```json5
module.exports = {
  root: true,
  env: {
    browser: true,
    mocha: true,
  },
  extends: [
    'plugin:vue/recommended',
    '@vue/airbnb',
  ],
  parserOptions: {
    parser: '@babel/eslint-parser',
    sourceType: 'module',
  },
  rules: {
    'arrow-parens': [2, 'always'],
    'import/extensions': 0,
    'import/no-cycle': 0,
    'import/no-extraneous-dependencies': 0,
    'import/no-unresolved': 0,
    'no-debugger': 1,
    'no-param-reassign': [
      2,
      {
        props: true,
        ignorePropertyModificationsFor: ['state'],
      },
    ],
  },
  overrides: [
    {
      files: [
        '*/components/**/*',
      ],
      rules: {
        'vue/sort-keys': 1,
      },
    },

    {
      files: [
        'src/**/*.spec.js',
        'tests/**/*.js',
      ],
      env: {
        jest: true,
      },
    },
  ],
};

```

**What did you do?**	
```js
// @vue/component
export default {
	computed: {
		panel() {
      		const fields = [
        		{		
					id: 'avatar',
          			component: 'field-avatar-picker',
          			entityType: 'user',
          			value: this.avatar,
				}
				// ...more fields
			];

			return {
				name : this.currentUser.name,
				fields,
			}
        },
	}
}

```

**What did you expect to happen?**
That my fields objects would be ignored

**What actually happened?**
```
149:11  warning  Expected object keys to be in ascending order. 'component' should be before 'id'                                           vue/sort-keys
```

**Comment by ota-meshi** [comment]
I didn't know what you thought was the problem. The rule seem to work fine.
If you don't like this rule, you can turn it off.
